symengine/pow.h

```diff
@@ -34,12 +34,12 @@ class Pow : public Basic
     //! \return `true` if canonical
     bool is_canonical(const Basic &base, const Basic &exp) const;
     //! \return `base` of `base**exp`
-    inline RCP<const Basic> get_base() const
+    inline const RCP<const Basic>& get_base() const
     {
         return base_;
     }
     //! \return `exp` of `base**exp`
-    inline RCP<const Basic> get_exp() const
+    inline const RCP<const Basic>& get_exp() const
```

Comment:
Sure, here are all the links: https://github.com/symengine/symengine/blob/f38254abd71543613e39588db40ee106d61cae46/doc/design.md#teuchos-rcp

In particular, this rule is on page 155 in http://www.cs.sandia.gov/%7Erabartl/TeuchosMemoryManagementSAND.pdf, in those tables.